src/lib.ts

```diff
@@ -0,0 +1,134 @@
+		child.on('error', reject);
+		child.on('exit', code => (code !== 0 ? reject(new Error(errorMessage)) : resolve()));
+	});
+}
+
+function createTask(callback: any) {
+	return new Promise((resolve, reject) => {
+		callback((error?: Error) => {
+			if (error) {
+				reject(error);
+			}
+			resolve();
+		});
+	});
+}
+
+function wrapUmd(content: string): string {
+	return `(function (root, factory) {
+if (typeof define === 'function' && define.amd) {
+	define([], function () { return (factory()); });
+} else if (typeof module === 'object' && module.exports) {
+	module.exports = factory();
+}
+}(this, function () {
+	return ${content};
+}));`;
+}
+
+function buildCss({ mode = 'dist', legacy }: any) {
+	const outDir = `output/${mode}`;
+	const cssFiles = path.join(basePath, outDir, '**/*.css');
+	const packageJsonPath = path.join(basePath, 'package.json');
+	const packageJson = fs.existsSync(packageJsonPath) ? require(packageJsonPath) : {};
+	const postcssPresetConfig = {
+		browsers: legacy ? ['last 2 versions', 'ie >= 10'] : ['last 2 versions'],
+		features: {
+			'nesting-rules': true
+		},
+		autoprefixer: {
+			grid: legacy
+		}
+	};
+
+	const modulesProcessor = postcss([
+		postcssImport(),
+		postcssPresetEnv(postcssPresetConfig),
+		postcssModules({
+			generateScopedName: mode === 'dist' ? '[hash:base64:8]' : '[name]__[local]__[hash:base64:5]',
+			getJSON: function(cssFileName: string, json: any) {
+				const newFilePath = cssFileName + '.js';
+				const themeKey = ' _key';
+				const packageName = packageJson.name;
+				json[themeKey] = `${packageName}/${path.basename(cssFileName, '.m.css')}`;
+				fs.writeFileSync(newFilePath, wrapUmd(JSON.stringify(json)));
+			}
+		}),
+		postcssCustomProperties({ preserve: false }),
+		postcssClean({ level: 2 })
+	]);
+	const variablesProcessor = postcss([
+		postcssImport(),
+		postcssPresetEnv(postcssPresetConfig),
+		postcssCustomProperties({ preserve: 'computed' })
+	]);
+
+	return globby(cssFiles).then(files => {
+		const moduleFiles = files.filter(file => file.endsWith('.m.css'));
+		const variableFiles = files.filter(file => !file.endsWith('.m.css'));
+		const processFile = (file: string) => {
+			const processor = file.endsWith('.m.css') ? modulesProcessor : variablesProcessor;
+			return processor
+				.process(fs.readFileSync(file, 'utf8'), {
+					from: file,
+					map: true
+				})
+				.then(({ css }: { css: string }) => {
+					fs.writeFileSync(file, css, 'utf8');
+				});
+		};
+		return Promise.all(moduleFiles.map(processFile)).then(() => Promise.all(variableFiles.map(processFile)));
+	});
+}
+
+export default function buildLibrary(args: any) {
+	const outDir = `output/${args.mode || 'dist'}`;
+	let tscOptions = ['--outDir', outDir];
+	if (!args.legacy) {
+		tscOptions = [...tscOptions, '-t', 'es6', '-m', 'esnext'];
+	}
+
+	const spinner = ora().start('Cleaning previous output');
+	return createTask((callback: any) => rimraf(outDir, callback))
```

Comment:
All these could be awaited also perhaps?

Reply:
I'll see what I can come up with, but personally I prefer the `then` series to a `try`/`catch` block in this particularly case.